### ISSUE TITLE
VS 2017 version

### ISSUE THREAD

**Issue by DejanMilicic** [zpqrtbnk/Zbu.ModelsBuilder#134]
When we could expect version for VS2017?

**Comment by larrynPL** [comment]
I'm also wondering when  we could expect Umbraco.ModelsBuilder.CustomTool for VS2017

**Comment by DejanMilicic** [comment]
@zpqrtbnk Could you please give us some rough timeframe for this? Thanx!

**Comment by zpqrtbnk** [comment]
Installed the RTM version of 2017 last week. Now I need to port the extension - which from what I am reading is not "immediate". We're working hard on putting together a new release of Umbraco at the moment, so I won't be able to do this extension in the coming 2-3 weeks.

The most reasonable timeframe would be approx 1 month, unless someone wants to try to do it.

**Comment by ruant** [comment]
Curses... 😄 
Maybe I should give it a stab this weekend.
Since I've some how messed up my VS2015 installation so it won't open up any of my projects.

**Comment by ruant** [comment]
@zpqrtbnk 
With some Iron Fist(ing) on Netflix as background noise I messed around a bit and managed to get it working.
Installs and tested it on a site running Umbraco 7.5.10

**Comment by DejanMilicic** [comment]
@ruant Mr.Robot is a more appropriate choice in this particular case :+1:

**Comment by zpqrtbnk** [comment]
Excellent, need to test!

**Comment by zpqrtbnk** [comment]
Quick Q re. the PR: I understand that it's now mandatory to build with VS 2017 (if we want to be able to target VS 2012 up to VS 2017) - is that correct?

And just because I don't have 2012 installed... you've tested the extension on 2012, right?

Planning a next minor release pretty soon and would love to merge!

**Comment by ruant** [comment]
@zpqrtbnk I have not tried building it with other then VS2017, but I guess VS2017 is need.
I can try it out in VS2015 later tonight, to see if it builds.

I have not tested it on 2012 no. The question is, do it need to support 2012?

I also haven't tested it in 2015 other then that the installer listed it as an option to install to during installation.
I'll check that it works with 2015 as well tonight.

**Comment by zpqrtbnk** [comment]
I don't think we need to support 2012, no, and I'm happy killing the support entirely.

If you can test that it works on 2015 then that'd be fantastic. And then I can merge and it'll be part of the next Core minor version.

Many, many, (many) thanks for helping here!

**Comment by ProNotion** [comment]
So glad you guys are working on this, I just forked the repo to have a go and then found this issue and the work you've done so far. This is the only reason I have to keep going back to VS2015 at the moment so looking forward to the next update. I have also read that you have to build VS2017 VSIX files only in VS2017.

**Comment by ruant** [comment]
@zpqrtbnk
Glad to help 👍 

Had a quick look at it yesterday.  
I didn't mannage to make it build in 2015.  
But I'm no expert, so it might be possible to get it to work.

Worth mentioning is that my VS2015 installation seams to be a bit screwed up after I installed VS2017.
So that might be the reason also.
Regarding testing the tool in VS2015 it's the same, VS2015 won't open up any of our projects, but my collegues manages. Same code 😕 

So if anyone can test out the VSIX built in VS2017, on VS2015 and confirm that it works, that would be awesome.
[Installer](http://95.85.19.139/Umbraco.ModelsBuilder.CustomTool.vsix)

**Comment by ProNotion** [comment]
I'll give it a go. 

> "All VSIX projects will require a major version one-way upgrade to Visual Studio 2017."

...which might explain why you can't now build it in VS2015. See https://docs.microsoft.com/en-us/visualstudio/extensibility/how-to-migrate-extensibility-projects-to-visual-studio-2017

**Comment by ProNotion** [comment]
@zpqrtbnk @ruant I have installed it and used it with no visible errors on the following:

Microsoft Visual Studio Enterprise 2015 (Version 14.0.25421.03 Update 3)
Microsoft Visual Studio Enterprise 2017 Version 15.0.26228.10 D15RTWSVC

I hope that helps!

**Comment by zpqrtbnk** [comment]
Thanks all for your helps, considering it all OK, will merge asap.

**Comment by zpqrtbnk** [comment]
Merged - note that the solution will open in VS2015, except for the VSIX project which will be marked as incompatible - VS2017 now required to build the whole thing.

A HUGE thank you to you all.

**Comment by zpqrtbnk** [comment]
Well I have an issue with the VSIX that is produced, the Visual Studio Gallery refuses it, and it cannot install on my machine, says that "the file is not a valid VSIX package" with the following log:
```29/03/2017 19:00:50 - Microsoft VSIX Installer
29/03/2017 19:00:50 - -------------------------------------------
29/03/2017 19:00:50 - vsixinstaller.exe version:
29/03/2017 19:00:50 - 15.0.26228.0 built by: D15REL
29/03/2017 19:00:50 - -------------------------------------------
29/03/2017 19:00:50 - Command line parameters:
29/03/2017 19:00:50 - C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VSIXInstaller.exe,D:\d\Zbu ModelsBuilder\build\Release\v3.0.6\Umbraco.ModelsBuilder.CustomTool-3.0.6.vsix
29/03/2017 19:00:50 - -------------------------------------------
29/03/2017 19:00:50 - Microsoft VSIX Installer
29/03/2017 19:00:50 - -------------------------------------------
29/03/2017 19:00:50 - Failed to initialize settings for Isolated Shell C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\ManagementStudio\ssms.exe: Impossible de trouver le point d'entrée '?' dans la DLL 'PkgdefMgmt.dll'.
29/03/2017 19:00:50 - Initializing Install...
29/03/2017 19:00:50 - Microsoft.VisualStudio.ExtensionManager.InvalidExtensionPackageException: The file is not a valid VSIX package.
   à Microsoft.VisualStudio.ExtensionManager.InstallableExtensionImpl.ReadVSIXManifestFromPackage(Stream stream, CultureInfo preferredCulture)
   à Microsoft.VisualStudio.ExtensionManager.InstallableExtensionImpl..ctor(String path, CultureInfo preferredCulture)
   à Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.CreateInstallableExtension(String extensionPath, CultureInfo preferredCulture)
   à Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.CreateInstallableExtension(String extensionPath)
   à VSIXInstaller.App.GetInstallableData(String vsixPath, Boolean isRepairSupported, IEnumerable`1& skuData)
   à VSIXInstaller.App.Initialize(Boolean isRepairSupported)
   à VSIXInstaller.App.Initialize()
   à System.Threading.Tasks.Task`1.InnerInvoke()
   à System.Threading.Tasks.Task.Execute()
```

No idea what it means exactly, reopening the issue...

**Comment by zpqrtbnk** [comment]
This is exactly [this issue](https://stackoverflow.com/questions/42891617/the-file-is-not-a-valid-vsix-package) and no, the VSIX does not contain `manifest.json` nor `catalog.json`. I've read the [migration notes](https://docs.microsoft.com/en-us/visualstudio/extensibility/how-to-migrate-extensibility-projects-to-visual-studio-2017) but it's not helping so far...

**Comment by zpqrtbnk** [comment]
Oops, it _may be_ that our build script still references VS2015... looking into it...

**Comment by zpqrtbnk** [comment]
Turns out it's the build script we're using to release MB that has issues. Fixed somehow, version 3.0.6 has been uploaded to NuGet and to Visual Studio Gallery. Should be part of next minor release of Cms (see [U4-9683](http://issues.umbraco.org/issue/U4-9683)).
